### ISSUE TITLE
does this work under ubuntu?

### ISSUE THREAD

**Issue by victoralex** [albertjuhe/camaleonX#1]
it seems to work only under windows

**Comment by albertjuhe** [comment]
Yes, this program has been developed in Java, the java language is multyplatform, works in windows and in linux. 
The most important think is that you have to have a Java 1.6 or higher. Another think is that you need to correct the documents path in the XML file files and write it in linux format.

**Comment by victoralex** [comment]
I've tried changing the paths in the xml config and into https://github.com/albertjuhe/camaleonX/blob/master/camaleonx.properties but it still tries with windows path separators "\" instead of unix "/"

**Comment by albertjuhe** [comment]
OK, let me try in a ubuntu so.

**Comment by albertjuhe** [comment]
Yes, I tried in an Ubuntu SO and this is a bug. Now ,only works in windows.
I correct this bug.

**Comment by victoralex** [comment]
Hi Albert,

Thank you for this. Do you know when i may be able to test it again on ubuntu?

Regards,

Victor

On July 3, 2015 at 14:12:07, Albert Juhe Brugué (notifications@github.com) wrote:

Yes, I tried in an Ubuntu SO and this is a bug. Now ,only works in windows.
I correct this bug.

**Comment by albertjuhe** [comment]
Yes, I will try to do that Monday.

2015-07-03 13:26 GMT+02:00 victoralex notifications@github.com:

**Comment by victoralex** [comment]
Yes, I will try to do that Monday.

**Comment by albertjuhe** [comment]
I've uploaded a new version with this path correction.
I've been testing the aplication in an Ubuntu SO, and works.

**Comment by victoralex** [comment]
great stuff :) will test it asap

**Comment by victoralex** [comment]
java -jar camaleonx.jar -U ./camaleonx.properties ./docx_merge_sample.xml -Xms256m -Xmx1024m -Dlog4j.configuration=./logs/camaleonx.log4j.xml -Dlog4j.debug=false -Dcamaleonx.local./home=logs

Reading config file: ./camaleonx.properties
CamaleonX v.0.1 Command line execution.
Processing ./docx_merge_sample.xml...
log4j:WARN No appenders could be found for logger (omaonk.xml.profile).
log4j:WARN Please initialize the log4j system properly.
Start merging [Docx]...
[Docx docx_merge_sample] Number of documents to merge: 2
[Docx docx_merge_sample]  Start merging... 
[Docx docx_merge_sample] Current document number: 1
[Docx docx_merge_sample] Merging: /Users/victorpopescu/Seeds/versions/1/components/camaleonX/./documents/1.docx
[Docx docx_merge_sample] Merge OK.
[Docx docx_merge_sample] Current document number: 2
[Docx docx_merge_sample] Merging: /Users/victorpopescu/Seeds/versions/1/components/camaleonX/./documents/2.docx
File : /Users/victorpopescu/Seeds/versions/1/components/camaleonX/./resources/tmp/docx_merge_sample\file1\ doesn't
[Docx docx_merge_sample] Merge Error.
[Docx docx_merge_sample] Removing temporal files./Users/victorpopescu/Seeds/versions/1/components/camaleonX/./resources/tmp/docx_merge_sample\
[Docx docx_merge_sample] Error merging KO.
[Docx docx_merge_sample] Merge Report:  
[Docx docx_merge_sample] Total merge time:00h 00m 00s 51ms, 2 documents merged.
[Docx docx_merge_sample] Merge ended DOCX'.
[Docx docx_merge_sample] Result file in:

**Comment by victoralex** [comment]
here is my camaleonx.properties file:

base=./resources/files/
results=./results/

#xsl docx path
app=./resources/
tmp=tmp/
xsl=xsl/
skeleton=skeleton/

tmp_pptx=tmp/
skeleton_pptx=skeleton_pptx/
xsl_pptx=xsl_pptx/

#Templates
templates=templates/

and my docx_merge_sample.xml:

<?xml version="1.0" encoding="UTF-8" ?>
<profile>
    <metas>
        <meta key="project_name" value="testdocx"/>
        <meta key="type" value="docx"/> 
    </metas>
    <contents>
          <content include="1" href="./documents/1.docx"/>
        <content  include="1" href="./documents/2.docx"/>

```
</contents>
```

</profile>

**Comment by victoralex** [comment]
i'm still getting trailing "/" using the latest version. i am using an osx

14.4.0 Darwin Kernel Version 14.4.0: Thu May 28 11:35:04 PDT 2015; root:xnu-2782.30.5~1/RELEASE_X86_64 x86_64

**Comment by albertjuhe** [comment]
Can you show me the camaleonX.properties file?

**Comment by victoralex** [comment]
On August 12, 2015 at 10:32:36, Albert Juhe Brugué (notifications@github.com) wrote:

Can you show me the camaleonX.properties file?

**Comment by chromakey** [comment]
Chiming in that I'm experience the same issue on OSX.  Here's my input/output after updating the properties file:

```
camaleonX master % java -jar camaleonx.jar -U camaleonx.properties docx_merge_sample.xml -Xms1024m -Xmx1024m -Dlog4j.configuration=file:/./logs/camaleonx.log4j.xml -Dlog4j.debug=true -Dcamaleonx.local./home=logs

Reading config file: camaleonx.properties
CamaleonX v.0.1 Command line execution.
Processing docx_merge_sample.xml...
log4j:WARN No appenders could be found for logger (omaonk.xml.profile).
log4j:WARN Please initialize the log4j system properly.
Start merging [Docx]...
[Docx docx_merge_sample] Number of documents to merge: 2
[Docx docx_merge_sample]  Start merging... 
[Docx docx_merge_sample] Current document number: 1
[Docx docx_merge_sample] Merging: /Users/chromakey/projects/camaleonX/documents/1.docx
[Docx docx_merge_sample] Merge OK.
[Docx docx_merge_sample] Current document number: 2
[Docx docx_merge_sample] Merging: /Users/chromakey/projects/camaleonX/documents/2.docx
File : /Users/chromakey/projects/camaleonX/resources/Users/chromakey/projects/camaleonX/tmp/docx_merge_sample\\file1\ doesn't
[Docx docx_merge_sample] Merge Error.
[Docx docx_merge_sample] Removing temporal files./Users/chromakey/projects/camaleonX/resources/Users/chromakey/projects/camaleonX/tmp/docx_merge_sample\
[Docx docx_merge_sample] Error merging KO.
[Docx docx_merge_sample] Merge Report:  
[Docx docx_merge_sample] Total merge time:00h 00m 00s 47ms, 2 documents merged.
[Docx docx_merge_sample] Merge ended DOCX'.
[Docx docx_merge_sample] Result file in: 
```

**Comment by victoralex** [comment]
so this is an osx issue? haven't tested on linux yet. Perhaps it will work on ubuntu directly. gonna try tomorrow

**Comment by chromakey** [comment]
I just ran it on an ubuntu server, same issue:

```
ubuntu@ip-172-31-13-254:~/camaleonX$ java -jar camaleonx.jar -U camaleonx.properties docx_merge_sample.xml -Xms1024m -Xmx1024m -Dlog4j.configuration=file:/./logs/camaleonx.log4j.xml -Dlog4j.debug=true -Dcamaleonx.local./home=logs
Reading config file: camaleonx.properties
CamaleonX v.0.1 Command line execution.
Processing docx_merge_sample.xml...
log4j:WARN No appenders could be found for logger (omaonk.xml.profile).
log4j:WARN Please initialize the log4j system properly.
Start merging [Docx]...
[Docx docx_merge_sample] Number of documents to merge: 2
[Docx docx_merge_sample]  Start merging... 
[Docx docx_merge_sample] Current document number: 1
[Docx docx_merge_sample] Merging: /home/ubuntu/camaleonX/./documents/1.docx
[Docx docx_merge_sample] Merge OK.
[Docx docx_merge_sample] Current document number: 2
[Docx docx_merge_sample] Merging: /home/ubuntu/camaleonX/./documents/2.docx
File : /home/ubuntu/camaleonX/./camaleonx/resources/./tmp/docx_merge_sample\\file1\ doesn't
[Docx docx_merge_sample] Merge Error.
[Docx docx_merge_sample] Removing temporal files./home/ubuntu/camaleonX/./camaleonx/resources/./tmp/docx_merge_sample\
[Docx docx_merge_sample] Error merging KO.
[Docx docx_merge_sample] Merge Report:  
[Docx docx_merge_sample] Total merge time:00h 00m 00s 37ms, 2 documents merged.
[Docx docx_merge_sample] Merge ended DOCX'.
[Docx docx_merge_sample] Result file in: 
```

**Comment by albertjuhe** [comment]
I'm installing  virtual Box, I have a snowleopart image, I will try over this osx.

**Comment by victoralex** [comment]
any luck?

**Comment by albertjuhe** [comment]
I can't install osx in the virtualbox, doesen't work.
In the ubuntu works.
![ubuntu](https://cloud.githubusercontent.com/assets/5848928/9309576/cd648eea-450a-11e5-9031-068b7be36e39.png)

I will recompile again the project and I will upload, because I dont understand the error.
The error with the slash changed, appear with file.getAbsolutePath(), has no sense, because this take the path from a file directly.
